CRM/Contact/BAO/Relationship.php

```diff
@@ -2112,17 +2118,29 @@ public static function getContactRelationshipSelector(&$params) {
           "action=view&reset=1&cid={$values['cid']}&id={$values['id']}&rtype={$values['rtype']}");
 
         if ($params['context'] == 'current') {
-          if (($params['contact_id'] == $values['contact_id_a'] AND $values['is_permission_a_b'] == 1) OR
-            ($params['contact_id'] == $values['contact_id_b'] AND $values['is_permission_b_a'] == 1)
+          if (($params['contact_id'] == $values['contact_id_a'] AND $values['is_permission_a_b'] == CRM_Contact_BAO_Relationship::EDIT) OR
+            ($params['contact_id'] == $values['contact_id_b'] AND $values['is_permission_b_a'] == CRM_Contact_BAO_Relationship::EDIT)
           ) {
             $relationship['sort_name'] .= '<span id="permission-a-b" class="crm-marker permission-relationship"> *</span>';
           }
 
-          if (($values['cid'] == $values['contact_id_a'] AND $values['is_permission_a_b'] == 1) OR
-            ($values['cid'] == $values['contact_id_b'] AND $values['is_permission_b_a'] == 1)
+          if (($params['contact_id'] == $values['contact_id_a'] AND $values['is_permission_a_b'] == CRM_Contact_BAO_Relationship::VIEW) OR
+            ($params['contact_id'] == $values['contact_id_b'] AND $values['is_permission_b_a'] == CRM_Contact_BAO_Relationship::VIEW)
+          ) {
+            $relationship['sort_name'] .= '<span id="permission-a-b" class="crm-marker permission-relationship"> +</span>';
```

Comment:
I see you're using `+` to denote the view-only permissioned relationships, corresponding with `*` for the view-and-edit permissioned relationships.  I wonder whether now might be the time to ditch those and use icons instead.

You could replace those spans with `<i class="crm-i fa-eye"></i>` for the view-only ones and then something similar for the view-and-edit ones.  I just happen to be unsure what is the best way to represent editing permission--`fa-pencil` is probably a bad choice because the user would want to click it thinking that's how you edit something.  I wonder what you (or others) think of the following:

- [fa-certificate](https://fontawesome.com/v4.7.0/icon/certificate) (it feels similar to the legacy asterisk, and it signifies that someone is given authority)
- [fa-pencil-square-o](https://fontawesome.com/v4.7.0/icon/pencil-square-o) (it denotes editing but is a different icon than we typically use for editing)
- [fa-user-circle-o](https://fontawesome.com/v4.7.0/icon/user-circle-o) (by showing a user in a circle, it might signify that they can edit the user)
- [fa-asterisk](https://fontawesome.com/v4.7.0/icon/asterisk) (we could always just replace a little asterisk with a big asterisk)

@colemanw might have thoughts on this too.